From e52ca76fe191eaaf8f8d6f80323df93fd2e0b3c0 Mon Sep 17 00:00:00 2001
From: Saurabh Sengar <saurabh.singh@xilinx.com>
Date: Fri, 2 Feb 2018 13:50:30 +0530
Subject: [PATCH 0254/1852] dt-bindings: display: xlnx: Adding sdi bindings

commit e08cc1469b0f138c81ed271ace301ce05faf0aab from
https://github.com/Xilinx/linux-xlnx.git

Adding device tree documentation of sdi transmitter subsystem.
Xilinx SDI transmitter subsystem implements a SDI transmit interface in
accordance to the serial digital interface (SDI) family of standards.
The subsystem accepts video from AXI-4 Stream Video interface and outputs
Native Video stream, and allows for fast selection of the top-level
parameters and automates most of the lower level parameterization. The
AXI4-Stream video interface allows a seamless interface to other
AXI4-Stream-based subsystems.

Signed-off-by: Saurabh Sengar <saurabh.singh@xilinx.com>
Reviewed-by: Hyun Kwon <hyun.kwon@xilinx.com>
Signed-off-by: Michal Simek <michal.simek@xilinx.com>
State: pending
Signed-off-by: Yaliang Wang <Yaliang.Wang@windriver.com>
---
 .../bindings/display/xlnx/xlnx,sdi-tx.txt     | 31 +++++++++++++++++++
 1 file changed, 31 insertions(+)
 create mode 100644 Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt

diff --git a/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt b/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt
new file mode 100644
index 000000000000..8015fe7afb7b
--- /dev/null
+++ b/Documentation/devicetree/bindings/display/xlnx/xlnx,sdi-tx.txt
@@ -0,0 +1,31 @@
+Device-Tree bindings for Xilinx SDI Tx subsystem
+
+The IP core supports transmission of video data in SDI Tx protocol
+
+Required properties:
+ - compatible: Should be "xlnx,sdi-tx".
+ - interrupts: Interrupt number.
+ - interrupts-parent: phandle for interrupt controller.
+ - reg: Base address and size of the IP core.
+ - port: Logical block can be used / connected independently with
+   external device. In the display controller port nodes, topology
+   for entire pipeline should be described using the DT bindings defined in
+   Documentation/devicetree/bindings/graph.txt.
+
+
+Example:
+
+	sdi_tx_subsystem@80000000 {
+		compatible = "xlnx,sdi-tx";
+		reg = <0x0 0x80000000 0x0 0x10000>;
+		interrupt-parent = <&gic>;
+		interrupts = <0 90 4>;
+		#address-cells = <1>;
+		#size-cells = <0>;
+		encoder_sdi_port: port@0 {
+			reg = <0>;
+			sdi_encoder: endpoint {
+				remote-endpoint = <&xyz_port>;
+			};
+		};
+	};
-- 
2.31.1

